src/EnrollOnPremController.js

```diff
@@ -35,9 +35,13 @@ function (Okta, FormType, FormController, Footer, TextBox) {
     },
     Model: function () {
       var provider = this.options.provider;
+      var factors = this.options.appState.get('factors');
+      var factor = factors.findWhere(_.pick(this.options, 'provider', 'factorType'));
+      var profile = factor.get('profile');
```

Comment:
If the factor is configured it will always send `profile`. However it's better to put an extra check in. Addressed this in my latest commit